### PR TITLE
[READY] Replaces the tesla/singulo on Pubby with a Supermatter

### PR DESCRIPTION
**Pull request by actioninja**
## About The Pull Request
As title.
Images in mapdiff for most recent update, here's one it rendered earlier:
image

~~I probably fucked up the map merge horribly I really don't know.~~ I didn't!

## Why It's Good For The Game

As a wise orange man once said, "it's way easier to balance one engine" or something like that I don't really remember.

## Changelog
:cl:
tweak: PubbyStation has been refitted from a Tesla and Singulo based engine to a Supermatter based engine.
/:cl:

Waiting on the donut supermatter mmm 👀

### CODE REVIEW

**Comment by actioninja** [comment]
I fucked up something with step_x and step_y I'll figure this out tomorrow.

**Comment by WJohn** [comment]
Open the map with notepad++ or something like that, find and replace all instances of step_x with pixel_xand all instances of step_y with pixel_y 

That's it.

**Comment by actioninja** [comment]
I fucked up worse than that somehow.

**Comment by zxaber** [comment]
Map merging is a pain in the ass and I'm confident it's the reason why so few people are willing to do mapping changes.

Anyway, I'm sad to see Pubby lose its Singlo, even if the SM _is_ better for the game.

**Comment by Farquaar** [comment]
While I understand why some want the singulo removed, I'm going to miss it. It really was the most fun engine, especially when it got loose. I'm still holding onto the hope that someone will balance it instead.

**Comment by actioninja** [comment]
Little do you all know the taser removal is just a smokescreen to get this merged without resistance.
If it were possible to plan on more dimensions, I would be.

**Review by ShizCalev — DISMISSED** [review]

**Comment by HomocidlePancakes** [comment]
Bad coder. Sheesh. Bad. Bad coder. BAD CODER!

**Comment by actioninja** [comment]
You having a stroke or something buddy?

**Comment by FloranOtten** [comment]
I truly enjoy the supermatter. I do. But we shouldn't make it a universal thing. A bit of variety is fun for engineers. 

I also think that repurposing the PA room into the emitter room is a stupid idea, and just looks majorly fucking goofy.

**Comment by actioninja** [comment]
> I truly enjoy the supermatter. I do. But we shouldn't make it a universal thing. A bit of variety is fun for engineers.
A bit of variety is fun if the singulo and tesla weren't trash engines that ruin way too many rounds early.

> I also think that repurposing the PA room into the emitter room is a stupid idea, and just looks majorly fucking goofy.
I was trying to go for a "poorly planned retrofit" vibe thematically on top of me not really wanting to remap a large section of it.

**Comment by plapatin** [comment]
dont take away the one good thing about pubby

**Comment by mrhugo13** [comment]
As much as the SM is flexible in it's design, i'd say this PR would remove one of the few unique things about Pubby.

Pubby is currently one of the two maps (not counting runtime silly) who doesn't use the SM as their main engine. if this were to be removed then Pubby would just have one of it's good features ripped out of it for the sake of streamlining setting up the SM.

One of the few problems I would see arising out of Pubbies engine is the fact that it's different engine can confuse new engineers who might've only recently played on the other 3 maps which uses the SM, potenially ending up with a released singulo or tesla by mistake.

However I could potenially see a new engine being uniquely made for Pubby as a better alternate solution to this, rather then just assimilating all stations into the ~~gllorious~~ SM hivemind, maybye stations should be given unique engines for the stations population standards and to give engineering something more unique to do then do SM setup No. 1 or SM *custom* setup No. 5235

TL;DR: Keep the Singulo/Tesla on pubby, or potenially make a new unique engine for Pubby only.

**Comment by actioninja** [comment]
I've said it a couple of times in this discussion thread, but the singulo and tesla fucking suck and the presence of them is an antifeature.  Pubby would be my favorite map if it didn't have the tesla ruining every other round on it.

**Comment by ShizCalev** [comment]
image
is this visible ingame?

looks janky as fuck from the render;
image

**Comment by actioninja** [comment]
> is this visible ingame?
> 
> looks janky as fuck from the render;

I couldn't figure out how to get it visible in game, it doesn't have a visible type like the others.

**Comment by Ratvar** [comment]
"if this were to be removed then Pubby would just have one of it's good features ripped out of it for the sake of streamlining setting up the SM".
Unique, not good. Big difference.

**Comment by PKPenguin321** [comment]
Singulo being "just sooo shit and bad you guys" is literally history revision. Yes it sometimes got loose early, no it really wasn't that big of an issue. Having it on one map is hurting nobody.

**Comment by zxaber** [comment]
Singlo/Tesla are just less interesting overall, really. There's not a lot you can experiment with, upgrade, or streamline like you can with the SM. While they're a bit iconic, and do make Pubby feel unique, the SM really is a better gameplay experience.

**Comment by PKPenguin321** [comment]
I don't know dude. I had lots of fun moments involving creative engine projects (quad singulos, disposals routed to singulo), silly setup mishaps (CE traps self in field/man gets emittered to death), tense fights in space where I'll try to stop somebody from sabotaging it, fun gimmicks involving the worship of the singulo...
I know this isn't the "is the singulo well designed?" thread but honestly much of what could be done with the SM can be done with it, only it's easier for players to understand and therefore they were more inclined to play with it and make fun situations. SM sabotages are always super subtle and lame while with the singulo a sabotage attempt was noticable right away and you could have actual conflict involving sec/engineering trying to prevent its escape. I am randomly interjecting this post so I can call out the maintainers who skimmed this post for not reading it when they inevitably merge the PR and say "but you had no good arguments not to". The SM being "mechanically better" because it takes 5 more minutes to delam, has the most boring engine failure effect of all time (hahaha wow engineering exploded again! So cool and fun haha) by default, and might turn into a singulo or a tesla sometimes if you do some atmos work always felt really over-exaggerated to me. The actual important bits of what makes the engine interesting mechanically are done just as well by the singulo.

**Comment by zxaber** [comment]
An engine's gameplay experience shouldn't be strictly limited to how sabotages work. That's part of it, sure, but not the extent.

When I say the SM is more interesting, I mean for the customization and experimentation you can do with it. Plasma engines, CO2 engines, prioritizing making Pluoxium over power, etc. Hell, I like to do a closed loop cooling system where plasma is cycled between heat exchange pipes in space and in the SM chamber, and the ~90/10 mix of CO2 and N2O is intentionally not scrubbed out.

The Singlo is neat, but it doesn't really offer that much except a routine list of steps to power the station. The multi-singlo thing is the closest to being a real custom setup, and you can still accomplish that by just ordering the parts. 

I _do_ wish we didn't have that silly containment thing when the SM fully delams, though. It was more interesting for the SM to be down to the wire, trying to fix it but knowing that if the integrity hit zero you were done for. The thirty seconds of grace period is a bit too hand-holdy (and the spam it throws out over common isn't that great ether).

**Comment by PKPenguin321** [comment]
Why on earth should that be the most important thing if it effectively does nothing to experiment like that? There's no difference between having 200kW in the grid or 2 million if both power the station, and all experimenting with the SM does is change that meaningless number.

**Comment by actioninja** [comment]
> Why on earth should that be the most important thing if it effectively does nothing to experiment like that? There's no difference between having 200kW in the grid or 2 million if both power the station, and all experimenting with the SM does is change that meaningless number.

So you're saying that the engine type is completely meaningless because power doesn't matter?
Good thing that balancing one engine's power output is a lot easier than balancing three.

**Comment by PKPenguin321** [comment]
No you illiterate, I'm saying that "experimenting" isn't even remotely the sole thing we should judge engines by. Singulo has a dozen other things going for it in other, more important areas. I know reading is hard for you but my big post up there goes over it pretty well.

**Comment by actioninja** [comment]
> No you illiterate, I'm saying that "experimenting" isn't even remotely the sole thing we should judge engines by. Singulo has a dozen other things going for it in other, more important areas. I know reading is hard for you but my big post up there goes over it pretty well.

Your big post is dumb, nanny nanny boo boo.

**Comment by PKPenguin321** [comment]
Christ.

**Comment by actioninja** [comment]
I'm in a pissy mood so I'm actually going to bother to give a proper response.

> Singulo being "just sooo shit and bad you guys" is literally history revision. Yes it sometimes got loose early, no it really wasn't that big of an issue. Having it on one map is hurting nobody.

How something was interacted with in the past is different than how it's going to be interacted with now. The playerbase changes, the attitudes change, a lot of things can change to where identical mechanics can become completely different in different contexts.
There also is the case of that now tesla and singulo only being on two maps (that are uncommonly played at that) has a sort of forbidden fruit effect where because it's not seen very often, so more people will want to release it just to watch since they haven't seen it in a while.

> I don't know dude. I had lots of fun moments involving creative engine projects (quad singulos, disposals routed to singulo), silly setup mishaps (CE traps self in field/man gets emittered to death), tense fights in space where I'll try to stop somebody from sabotaging it, fun gimmicks involving the worship of the singulo...

This is where I take the most issue with your entire premise because the argument shifts from "this engine is fine" to "well I have fun with it so it must be fine."
You can route disposals to the SM crystal, there is a huge amount more mishaps that can happen in setup, fights over sabotage can happen as well, just from a pure numbers game the SM wins out over the singulo and tesla. However, reality isn't a numbers game and objectively having more features and variables doesn't make the SM better. What does make it better is that those variables and features are better thought out and make for a more engaging player experience.
Instead of an easy sabotage that stopping requires constant attention towards a thing that otherwise does absolutely nothing, there's a lot of different fairly complicated routes of sabotage that making it difficult to fix requires a bit of knowledge about how the engine works.
In fact, that's the reason why I think the SM is a fundamentally _much_ better design than any of the other engines. While basic interaction with the SM can be done by anyone willing to read a couple paragraph guide, advanced interaction such as optimization, repair of a complicated sabotage, and elaborate delams _demand_ knowledge of how it actually works. It's not straightforward, it's a convoluted mess with so many points of tweaking and variability that it rewards intricate knowledge of the system.
Meanwhile the tesla and singulo are the opposite. There is nearly nothing deeper to understand; if containment is setup in the one proper way, it's safe, you sabotage it by destroying the containment, and then it will wreck everyone's shit whether or not the person destroying it knew anything about what it did or how it works besides "shit's gonna go real bad real soon."

And then just to be an asshole:

> Why on earth should that be the most important thing if it effectively does nothing to experiment like that? There's no difference between having 200kW in the grid or 2 million if both power the station, and all experimenting with the SM does is change that meaningless number. The SM is just as much of a flowchart to set up, too.

Weren't you just defending that the singulo and tesla are just as good because they totally have room for experimentation too? But now experimentation doesn't matter because it's just a meaningless number anyways? So which is it, are they equal because they both have room for experimentation or does the SM really have no advantage because experimentation is pointless?
And sure, a basic SM setup is a simple step by step process, but if you want to do something weird and experimental, there's a ridiculous amount of options you have due to the atmos integration. Changing the order of the cooling and filtering, using weird gas mixes, number of emitters, bringing in the freezers, using dead bodies for radiation feedback, which already is more than "you can put MORE singulo/tesla generators in the containment!" despite not being an exhaustive list.

**Comment by PKPenguin321** [comment]
I'd like to say that I genuinely appreciate your real response.

>Weren't you just defending that the singulo and tesla are just as good because they totally have room for experimentation too? But now experimentation doesn't matter because it's just a meaningless number anyways? So which is it, are they equal because they both have room for experimentation or does the SM really have no advantage because experimentation is pointless?

I never said experimentation is worthless, just that it is the least valuable metric to judge an engine by (again, this is in regards to "experimentation" where it is defined as maximizing power output, which is worthless since a powered station is a powered station). Experimentation is definitely a good thing, but the other factors at play when weighing engines against each other are definitely of a higher priority, especially ones that actually do something (again, because going from 200kW to 2 million is effectively doing nothing). It's not an argument of exclusivity, but of priority and importance.
And just to be a bit of a sarcastic asshole, what you listed as "better" experimentation sounds kind of lame to me. "You can put WEIRD gas mixes! You can put MORE freezers!" Wow, thrilling.

>You can route disposals to the SM crystal, there is a huge amount more mishaps that can happen in setup, fights over sabotage can happen as well, ...

I'll give you that. I've seen both of these things happen. The reason I even brought it up is because freeing the singularity is often looked at as "WOW SO EASY NO CHANCE TO FIGHT BACK," which I mean, it's definitely easier than blowing the supermatter (mostly because there's so many layers of handholding before the SM goes off), but I wanted to show how fights did happen over singuloosing and that it wasn't always just "anybody can loose it with no effort!!"

>While basic interaction with the SM can be done by anyone willing to read a couple paragraph guide, advanced interaction such as optimization, repair of a complicated sabotage, and elaborate delams _demand_ knowledge of how it actually works.

This is the part that I think is always over-exaggerated with the SM. "Complicated sabotage" invariably results in the SM going off because nobody dares to fuck with it or the guy who sabotaged it is also there defending it. *In theory,* I agree it could happen, but it never does. Most sabotage fixing is just "wrench the SM-not-explode gas into the input connector" and has next to no depth.
"Optimization" is just another word for "pumping the wattage output" which is the same as doing nothing once the station is being energized. I've gone over this already so I'll move on.
Elaborate delams, I'll give you. Those are interesting, but I wish they actually happened more often. They take so long that very few antags will give them the time investment, and often the shuttle is called before much damage is done, but when they do work out it's pretty cool.

>Meanwhile the tesla and singulo are the opposite. There is nearly nothing deeper to understand; if containment is setup in the one proper way, it's safe, you sabotage it by destroying the containment, and then it will wreck everyone's shit whether or not the person destroying it knew anything about what it did or how it works besides "shit's gonna go real bad real soon."

Two things wrong with this:
- You could definitely reinforce and secure the singulo containment area. Hacking the PA to lock it down, upgrading cameras so the AI can keep better watch, putting up more walls, shocking doors, putting redundant emitters, setting extra midpoints for the field generators, I even saw some people set up backup PACMAN generators for the emitters a handful of times with the particularly fort-building-centric engineer players. Containment had more than one proper way of doing it beyond the bare minimum.
- The SM, when set up to the bare minimum, is also easy for a player with even trivial knowledge of the SM to sabotage. This isn't exclusive to the singulo.

>This is where I take the most issue with your entire premise because the argument shifts from "this engine is fine" to "well I have fun with it so it must be fine."

Perhaps I phrased it to sound that way, but you should get what I meant by it. There were fun things to do and there was fun to be had with the singulo. I had fun, yeah, but others did too, and me individually having fun is definitely not the crux of my argument. Assuming that it is is just facetious.

>There also is the case of that now tesla and singulo only being on two maps (that are uncommonly played at that) has a sort of forbidden fruit effect where because it's not seen very often, so more people will want to release it just to watch since they haven't seen it in a while.

This is the type of thing that should be quantified with evidence. In theory, sure. But even then, wouldn't leaving it in alleviate this effect if it's being caused by rarity in the first place?

I know this is pointless since oranges will just merge it without a second thought, but I do appreciate you giving an actual reply and decided it was worth a real reply in exchange. No need to be in a pissy mood, I'm not trying to shit on you or anything. I genuinely believe that the singulo has merit as an engine and that it causes no harm staying on one map of our many. I haven't even touched on how variety is a boon, and just having different engines on different maps provides that. There are reasons that the singulo should stay on this one map, and there are reasons that the SM should stay on all the others. I'm certain that there are significant swathes of players on the servers right now that enjoy having the singulo instead of the supermatter just now and again, and at the end of the day, the collective enjoyment of the game that the playerbase experiences is what is really most important when it comes to designing a video game, because this is all just that: A video game.

If you really do read this whole post, I'd like to thank you for letting me spill my thoughts. You don't have to agree with me, but it's nice to have my beliefs acknowledged either way. So yeah, thanks.

**Comment by FloranOtten** [comment]
Just want to add in fighting over the SM BARELY happens outside shit like nukies using the nearby airlock to enter. Most traitors (like me) will just pretend to fix it, wrench in a can of something (usually oxygen repainted as n2o) and keep fucking with the settings because noone can see what you're doing. I have seen 2 proper battles of traitors vs crew for the SM delam, and both ended with the SM being saved because the traitor walked into an anomaly.

**Comment by FloranOtten** [comment]
Another thing; This attempt to fit a standard SM (I presume you took the design from Delta as that's the only SM with heaters and a special filter for them) looks so fucking bad. Maybe remove that freezer backloop thing and remove the bulge from the plasmaglass, and place the freezer loop there?

**Comment by optimumtact** [comment]
If all of you rabid singulo and tesla fans had ever bothered either fixing the tesla or making the singularity actually more interesting then this conversation might have had a merit.

But as it is, the only engine with anything approaching past skin deep interaction is the SM and I'm not gonna wait around for 300 years just because of history.

**Comment by datorangebottle** [comment]
Why was this engine merged in this state? The coolers on the right aren't connected roundstart. The emitters can't just be turned on like meta's setup. There's a layer manifold underneath the floor on the waste loop. There are HEATERS on the COOLING loop, as if it wasn't easy enough to sabotage. Finally, why the fuck is the heat exchange loop so massive?

image
image


This is the worst roundstart engine I have ever seen, and I pray we never have to see another like it.

**Comment by PKPenguin321** [comment]
https://youtu.be/2e5tAlelJXc?t=42

**Comment by Cartlord** [comment]
> 
> 
> If all of you rabid singulo and tesla fans had ever bothered either fixing the tesla or making the singularity actually more interesting then this conversation might have had a merit.
> 
> But as it is, the only engine with anything approaching past skin deep interaction is the SM and I'm not gonna wait around for 300 years just because of history.

Did you just... miss this entire comment? ( https://github.com/tgstation/tgstation/pull/42932#issuecomment-468435443 )

**Comment by optimumtact** [comment]
move on man it was a year ago
